### PR TITLE
Tenant DB error handling

### PR DESCRIPTION
**Pull request by pdeziel**
### Scope of changes

This updates the Tenant endpoint handlers to return 404s when Trtl returns NotFound and fixes the "update" endpoints so that the updates actually occur.

This also lays some groundwork for the organization ID checks.

Fixes SC-13543

### Type of change

- [ ] new feature
- [ ] bug fix
- [ ] documentation
- [ ] testing
- [x] technical debt
- [ ] other (describe)

### Acceptance criteria

Most important thing to look at are the update routines.

### Author checklist

- [x] I have manually tested the change and/or added automation in the form of unit tests or integration tests
- [ ]  I have updated the dependencies list
- [ ]  I have recompiled and included new protocol buffers to reflect changes I made
- [ ]  I have added new test fixtures as needed to support added tests
- [x]  Check this box if a reviewer can merge this pull request after approval (leave it unchecked if you want to do it yourself)
- [x]  I have moved the associated Shortcut story to "Ready for Review"

### Reviewer(s) checklist

- [ ] Any new user-facing content that has been added for this PR has been QA'ed to ensure correct grammar, spelling, and understandability.
- [ ] Are there any TODOs in this PR that should be turned into stories?

### CODE REVIEW

File: pkg/tenant/apikeys.go [hunk 1]
```diff
@@ -33,6 +32,7 @@ func (s *Server) ProjectAPIKeyList(c *gin.Context) {
 	}
 
 	// orgID is required to check ownership of the project
+	// TODO: Ensure that the project is in the organization
```
Comment (bbengfort) [inline pkg/tenant/apikeys.go:35]: Do we have a security problem with this until we fix the project check?

**Review by bbengfort — APPROVED** [review]
🚀

File: pkg/tenant/apikeys.go (hunk 1 above)
Comment (pdeziel) [reply pkg/tenant/apikeys.go:35]: Yes we'd have to retrieve the project to do the check, should I add that back in until we have an alternative way to perform the check?

Comment (pdeziel) [reply pkg/tenant/apikeys.go]: I added the checks back in since List/Create API keys is a crucial part of the beta release.

**Pull request merged**